src/server/req_rescq.c

```diff
@@ -583,7 +583,9 @@ req_confirmresv(struct batch_request *preq)
 	 */
 	if (is_being_altered)
 		free_resvNodes(presv);
-	rc = assign_resv_resc(presv, next_execvnode);
+	int svr_init;
+	svr_init = FALSE;
+	rc = assign_resv_resc(presv, next_execvnode, svr_init);
```

Comment:
You do not need extra variable here. Directly pass the desired value to the function assign_resv_resc.

Reply:
> I see a problem with the code. Standing reservations move from occurrence to occurrence when one finishes. If a vnode goes down (or become stale) between when the reservation is confirmed and when the occurrence is the next occurrence, we'll call set_nodes() with down/stale nodes. The new change will reject that change and the reservation be left in an unstable state. The reservation will lose its vnodes for the next occurrence.
> Right now we at least set the vnodes (since we ignore state), but we don't degrade the reservation.
> The set_nodes() function is an extremely critical function and some parts of the code don't handle when it fails well.
> 
> How about this as a solution:
> Handle reservations differently than jobs. The current code works fine for jobs. Other than server restart, all calls to set_nodes() for jobs are from requests. We can reject those. For reservations, we detect but do not reject. When we detect the down/stale situation, we correctly reallocate all the nodes to the reservation, but immediately degrade it. The scheduler will reconfirm it on other nodes later.
> 
> Another idea is slightly different. We can reject a reservation confirm, but on other calls to set_nodes() we detect and degrade.
> 
> What do you think?

Thanks Bhroam - Made changes for code to take care of Normal reservation and Standing reservation as well.
In case of normal reservation added retry after 10 seconds, as the nodes may be available - if the call has come to set nodes.
Standing reservation - I've set it for degraded, which will be considered for next scheduling cycle.

please check the changes and let me know the comments.